### ISSUE TITLE
transport (and other function in config) are broken

### ISSUE THREAD

**Issue by zenoxs** [mowispace/react-native-logs#108]
Looks like a big regression has been introduced by https://github.com/mowispace/react-native-logs/pulls?q=is%3Apr+is%3Aclosed 

By doing a `JSON.parse(JSON.stringify(config))` every function defined in the config is transformed into a null value.

```js
const transport = [
  () => {
  },
  () => {
  },
];
JSON.parse(JSON.stringify(transport)) // output [null, null]
```

If you wanna do a deepmerge you'd probably better to use a lib such as https://www.npmjs.com/package/deepmerge-ts or write a custom function

**Comment by damaestro** [comment]
This is causing the error `ERROR  Error: [react-native-logs] ERROR: transport is not a function, js engine: hermes` to be shown and logging is broken.

```
const defaultConfig = {
  levels: {
    debug: 0,
    info: 1,
    warn: 2,
    error: 3,
  },
  severity: __DEV__ ? "debug" : "error",
  transport: [
    consoleTransport,
    sentryTransport,
  ],
  transportOptions: {
    colors: {
      info: "blueBright",
      warn: "yellowBright",
      error: "redBright",
    },
    SENTRY: Sentry,
    errorLevels: ["fatal", "error", "warning"],
  },
  async: true,
  asyncFunc: InteractionManager.runAfterInteractions,
  dateFormat: "time",
  printLevel: true,
  printDate: true,
  fixedExtLvlLength: false,
  enabled: true,
};
```

The following gets us back to working:
```
npm install react-native-logs@5.1.0
```

**Comment by alessandro-bottamedi** [comment]
Sorry, I hadn’t considered the functions. For now, let’s roll back and keep the old config merging system.

Fixed in v 5.2.2.
